widgetssdk/src/main/java/com/glia/widgets/GliaWidgets.java

```diff
@@ -219,12 +219,13 @@ public static void clearVisitorSession() {
      */
     public static void endEngagement() {
         Logger.d(TAG, "endEngagement");
+        Dependencies.getControllerFactory().destroyControllers();
+
         Dependencies.glia().getCurrentEngagement().ifPresent(engagement -> engagement.end(e -> {
             if (e != null) {
                 Logger.e(TAG, "Ending engagement error: " + e);
             }
         }));
-        Dependencies.getControllerFactory().destroyControllers();
```

Comment:
Does the sequence of calls matter here? I'm asking because there is a risk that changed sequence can cause other issues which we are not aware of.

Reply:
I think yes, it matters because destroying controllers before calling the `endEngagement` guarantees that all the `end engagement` subscribers will be unsubscribed before the `end-engagement` event is received, so we will be safe from leaking the survey to the next engagement and so on.
I want to remind you that integrators are using this method to end engagement outside of the Widgets SDK app, so we need to clean all the data(surveys included), because this is usually used when integrators log out users from the app.

Reply:
cc: @Pelkar @AndriiHorishniiMOC for visibility